### PR TITLE
feature: Autoplay the next video after completion of one #1308

### LINKED ISSUE #1308: feature: Autoplay the next video after completion of one
**Is your feature request related to a problem? Please describe.**
It is a hassle when the video ends and you have to go to contents tab and play the next video. Not sure if it's just me facing this issue but it is annoying

**Describe the solution you'd like**
I would want my next course video to be autoplayed whenever the present video ends. Just like the Youtube autoplay

**Describe alternatives you've considered**
Or maybe have a button in the corner which on clicking it redirects to the next video

I feel like it's doable for me, would like to work on this feature enhancement

### PR DESCRIPTION
**Pull request by SahilLamba0008**
### PR Fixes:
- 1 Autoplay Next Video: Implemented functionality to autoplay the next video after one completes.
- 2 Minor Code Cleanup: Refactored some code related to the video player to ensure smooth transitions between videos.

Resolves #[Issue Number if there] 
- Resolves #1308

https://github.com/user-attachments/assets/0b164fba-28c2-4986-b59f-a41f9afeb9d2

### Checklist before requesting a review
- [ ✅ ] I have performed a self-review of my code
- [ ✅ ] I assure there is no similar/duplicate pull request regarding same issue

### CODE REVIEW

**Comment by devsargam** [comment]
Adding it to things to reivew

**Comment by devsargam** [comment]
Mean while if u can clear the conflicts @

**Comment by SahilLamba0008** [comment]
> Mean while if u can clear the conflicts @

I've resolved the conflicts in the PR, Please review and let me know if there are any other changes needed.

**Comment by Prashant-flick** [comment]
@SahilLamba0008 
I think this will only work in a week, it will not play the next week video, can you fix that

**Comment by SahilLamba0008** [comment]
@Prashant-flick @devsargam 
Thank you for the feedback! While I understand the request to have videos automatically continue into the next week's module, I believe this approach may unnecessarily increase the load, especially since not all users may be ready to proceed to the next week's content right away.

Instead, I propose implementing a modal at the end of the last video in the current week's module, similar to YouTube's playlist behavior. This modal could prompt users with an option like "Continue to next week's videos" (refer to image for more context), giving them control over when they want to move forward. This would provide a smoother user experience and reduce potential server or client load by avoiding preloading content until the user chooses to proceed.

Let me know if this sounds like a good compromise!

Screenshot 2024-10-09 at 3 53 46 PM

**Comment by SahilLamba0008** [comment]
> @SahilLamba0008 whatever you are thinking is somewhat correct but I would need a bit more context Can you share a loom https://www.loom.com/ with your explanation

check this out ( https://www.loom.com/share/5be7066dc3ac4d81b1b1a7f7df72b7ae?sid=401e93af-557b-4e43-8c6c-1defd395dc84 ). I'm suggesting to add a modal just like udemy does, suggesting next weeks content on completion of the last video of current week. This modal can have text as `Up Next` and `Next Weeks Module Name` with a play button.

File: src/db/course.ts
Comment (devsargam) [inline src/db/course.ts:166]: You have a lot of console.log comments. That's really bad. Could you please resolve those?

File: src/components/admin/ContentRendererClient.tsx
Comment (devsargam) [inline src/components/admin/ContentRendererClient.tsx:84]: here

File: src/components/admin/ContentRendererClient.tsx
Comment (devsargam) [inline src/components/admin/ContentRendererClient.tsx:75]: ?

**Review by devsargam — CHANGES_REQUESTED** [review]

File: src/components/admin/ContentRendererClient.tsx
Comment (SahilLamba0008) [reply src/components/admin/ContentRendererClient.tsx:75]: yes, on it.

**Comment by devsargam** [comment]
Also, as a follow up pr. Would you be interested to add `next` button in the video player itself

**Comment by SahilLamba0008** [comment]
> Also, as a follow up pr. Would you be interested to add `next` button in the video player itself

yeah sure, do i need to add prev button as well just like a proper naviagtion within the player itself.

**Comment by SahilLamba0008** [comment]
> > Also, as a follow up pr. Would you be interested to add `next` button in the video player itself
> 
> @devsargam can you please check if these are the intended changes.
> 
> * I've made changes keeping in mind so that user can only navigate through contents of `video` type.
> 
>  Screen.Recording.2024-10-25.at.8.26.17.PM.1.mp4

@devsargam changes are made in this commit "[Added navigation buttons within videojs controller with autoplay feature](https://github.com/code100x/cms/pull/1434/commits/6ab6eb74e1c7f7ae2574675109b671028b9f5816)" please review.

**Comment by SahilLamba0008** [comment]
@devsargam Is this okay or do I've to raise a new issue for this.

File: prisma/seed.ts
Comment (devsargam) [inline prisma/seed.ts:184]: remove

Comment (SahilLamba0008) [reply prisma/seed.ts:184]: on it

Comment (SahilLamba0008) [reply prisma/seed.ts]: done

**Comment by SahilLamba0008** [comment]
@devsargam Please review, I guess it's good now.

File: prisma/seed.ts
Comment (SahilLamba0008) [reply prisma/seed.ts]: there are several other consoles which are not added by me, I can remove them as well if you want.

**Comment by SahilLamba0008** [comment]
@devsargam Any updates on the review ?

**Comment by SahilLamba0008** [comment]
hi @devsargam, it seems you are quite busy. Are there any other maintainers I could ask for a review?

**Comment by SahilLamba0008** [comment]
hi @devsargam, should i open a new issue for this pr ?

**Comment by devsargam** [comment]
Let me review this one sir. Further more I will inform. I had to take a break so currently getting everything in line

**Comment by SahilLamba0008** [comment]
> Let me review this one sir. Further more I will inform. I had to take a break so currently getting everything in line

Apologies sir! Please take your time. I don't mean to rush you.

**Comment by SahilLamba0008** [comment]
Hi @devsargam, I've resolved the latest conflicts. Still waiting for your review—please take a look whenever you're free. I'm up for any further tweaks required.

**Comment by devsargam** [comment]
I will review it today if possible 
thanks for waiting

**Comment by devsargam** [comment]
image

**Comment by devsargam** [comment]
Works good on dev but breaks during the build might want to fix it @SahilLamba0008

**Comment by SahilLamba0008** [comment]
> Works good on dev but breaks during the build might want to fix it @SahilLamba0008

@devsargam I assume the feature is working as expected—please confirm. I'll address the build error ASAP.

**Comment by devsargam** [comment]
Feature works yeah for a week it does but not after that which is fine. You did add the buttons so that should help us implement later on easily so yeah good work

**Comment by devsargam** [comment]
@SahilLamba0008 can you resolve them by today?

**Comment by SahilLamba0008** [comment]
> @SahilLamba0008 can you resolve them by today?

Yeah, already done. Will raise a PR when I reach home, probably by tonight.

**Comment by SahilLamba0008** [comment]
@devsargam All set, bro. I've run the build command locally as well, just to be sure.

**Review by devsargam — APPROVED** [review]

**Comment by SahilLamba0008** [comment]
@devsargam It's been 2 weeks since you approved the changes, could you let me know how long it might take to get it merged, or if I need to resolve any conflicts on my end ?

**Comment by devsargam** [comment]
nothing is wrong. there are important stuff to look at atm that's what holding it back. 
if you are curious vidoes not playing with the new appx support

**Comment by devsargam** [comment]
@SahilLamba0008 deprioritizing this seems a bit flaky imho. You did a great job but there are a lot of conflicts and this might cause issues on prod. closing. Sorry for wasting your time but feel free to pick any other issues

**Pull request closed without merging**
